language/stdlib/modules/Libra.move

```diff
@@ -778,7 +769,7 @@ module Libra {
         currency_code: vector<u8>,
     ): (MintCapability<CoinType>, BurnCapability<CoinType>)
     {
-        assert(Roles::has_register_new_currency_privilege(lr_account), Errors::requires_role(EREGISTRATION_PRIVILEGE));
+        Roles::assert_libra_root(lr_account);
```

Comment:
I used "has_register_new_currency_privilege" because Sam and Tim recommended that I do so.  I'm happy to get rid of it, but you should make sure Sam is ok with doing that.

Reply:
I asked Junkil actually to do this. The older guidance became obsolete after the Errors PR and Sam and Tim also prefer now those assert methods, which we already have introduced most places in the framework.

Reply:
I'm happy to see this change since we don't use explicit privileges anywhere else outside of capabilities, so I think it's best to move everything over to be role-based for now until we get proper capabilities to represent privileges for consistency. 

Agreed with Wolfgang that since the new errors framework, assertions from the module that defines the error condition is the preferred method.